### PR TITLE
feat: add IAM samples

### PR DESCRIPTION
**Pull request by sethvargo**
/cc @bcoe

### CODE REVIEW

File: samples/test/secretmanager.test.js [hunk 1]
```diff
@@ -23,7 +23,8 @@ const client = new SecretManagerServiceClient();
 
 const projectId = process.env.GCLOUD_PROJECT;
 const secretId = uuidv4();
-const payload = `my super secret data`;
+const payload = 'my super secret data';
+const iamUser = 'user:sethvargo@google.com';
```
Comment (grayside) [inline samples/test/secretmanager.test.js]: 👋 I don't think this repo has much existing facilitate to help, but this should probably be an environment variable.

Comment (sethvargo) [reply samples/test/secretmanager.test.js]: @bcoe advise here? We can use an envvar and then set something in the koroko config. Can you make me a service account on the test project please?

Comment (bcoe) [reply samples/test/secretmanager.test.js]: we don't currently have a great mechanism for setting environment variables unfortunately, I'd be comfortable with this hack for now, but I suggest we address this shortcoming in our CI/CD, as this has come up a few times recently.

I'm picturing that we populate environment variables in a well known file in GCS potentially, and then inject them into the test environment.

Comment (grayside) [reply samples/test/secretmanager.test.js]: Maybe it's worth copying what nodejs-docs-samples does? [Non-secrets are managed in a shell script](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/master/.kokoro/build.sh) as part of the repo for ease of config management.

Secrets are loaded from a directory brought in by Kokoro config, but that doesn't seem very different from using a storage bucket.

Comment (sethvargo) [reply samples/test/secretmanager.test.js]: I like what Go does (similar to what Adam linked) - just a big env file.

Comment (bcoe) [reply samples/test/secretmanager.test.js]: I created a tracking issue, we do have shared config that gets updated nightly [here](https://github.com/googleapis/synthtool/blob/master/synthtool/gcp/templates/node_library/.kokoro/samples-test.sh), but it doesn't currently create usable commit messages on repos (so it creates a lot of work whenever we have an environment variable).

I'd like to address the problem of rolling out environment variable updates in a more holistic way, and think we should go ahead and land this as it stands.

**Pull request merged**